### PR TITLE
Implement transaction authorization

### PR DESCRIPTION
**Pull request by tuommaki**
Gevulot devnet is permissioned due to its nature. For this need,
implement basic transaction authorization based on authors' public keys.

### CODE REVIEW

File: crates/node/migrations/20231128120351_transactions-table.sql [hunk 1]
```diff
@@ -12,6 +12,7 @@ DROP TYPE IF EXISTS transaction_kind;
 CREATE TYPE transaction_kind AS ENUM ('empty','transfer', 'stake', 'unstake', 'deploy', 'run', 'proof', 'proofkey', 'verification', 'cancel');
 
 CREATE TABLE transaction (
+    author VARCHAR(130) NOT NULL,
```
Comment (tuommaki) [inline crates/node/migrations/20231128120351_transactions-table.sql:15]: This is a bit drive-by change and I wouldn't normally combine these, but gotta move fast :see_no_evil:

File: crates/node/src/main.rs [hunk 2]
```diff
@@ -140,22 +153,20 @@ impl workflow::TransactionStore for storage::Database {
     }
 }
 
-struct AuthenticatingTxHandler {
```
Comment (tuommaki) [inline crates/node/src/main.rs:143]: Authorizing in `mempool` is better approach than in P2P handler. `mempool` captures both, JSON-RPC & P2P.

File: crates/node/src/main.rs [hunk 3]
```diff
@@ -76,14 +76,27 @@ async fn main() -> Result<()> {
...
+                db.acl_deny(&key).await
             }
         },
         Command::Run { config } => run(Arc::new(config)).await,
+        Command::Show { op } => match op {
+            ShowCommand::PublicKey { key_file } => {
+                let bs = std::fs::read(key_file)?;
+                let key = SecretKey::parse(bs.as_slice().try_into()?)?;
+                let public_key = PublicKey::from_secret_key(&key);
+                println!("{}", hex::encode(public_key.serialize()));
```
Comment (musitdev) [inline crates/node/src/main.rs:96]: Use log::trace! instead of println!

**Review by musitdev — APPROVED** [review]
It opens a lot of discussion about software architecture, auth needs, ... but as you say we've to move fast. So we'll see later.

File: crates/node/src/main.rs (hunk 3 above)
Comment (tuommaki) [reply crates/node/src/main.rs:96]: Actually, isn't `println!` totally fine in this case as it's purely meant to be "write to stdout" for end user, no matter what the logging configuration is?

**Comment by tuommaki** [comment]
> It opens a lot of discussion about software architecture, auth needs, ... but as you say we've to move fast. So we'll see later.

True, but to make it even funnier, once we get into testnet phase, the authorization goes away :stuck_out_tongue_closed_eyes:

File: crates/node/src/main.rs (hunk 3 above)
Comment (musitdev) [reply crates/node/src/main.rs:96]: Ah ok, I didn't really investigate the reason, I thought it was one forgets.
